app/views/admin/locations/_form.html.erb

```diff
@@ -1,46 +1,68 @@
-<%= form_for location, url: admin_location_path do |f| %>
-  <div class="form-group">
+<%= form_for location, url: admin_location_path, method: 'put' do |f| %>
+  <% if location.errors.any? %>
+    <div class="alert alert-danger t-errors" role="alert">
+      <h2 class="alert__heading">There's a problem</h2>
+      <ul>
+        <% error_messages_for(location, location.address).each do |attribute, attribute_errors| %>
+          <% attribute_errors.each do |attribute_error| %>
+            <li class="t-error-message"><a href="#<%= attribute %>"><%= sanitize(attribute_error, :tags => []) %></a></li>
+          <% end %>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
+
+  <div class="form-group" id="title">
     <%= f.label :title %>
+    <%= render 'error', error_messages: location.errors[:title] %>
     <%= f.text_field :title, class: 'input-md-3 form-control t-location-title' %>
   </div>
 
   <%= f.fields_for(:address, location.address) do |a| %>
-    <div class="form-group">
+    <div class="form-group" id="address_line_1">
       <%= a.label :address_line_1, 'Address' %>
+      <%= render 'error', error_messages: location.address.errors[:address_line_1] %>
       <%= a.text_field :address_line_1, class: 'input-md-3 form-control t-address-line-1' %>
     </div>
 
-    <div class="form-group">
+    <div class="form-group" id="address_line_2">
+      <%= render 'error', error_messages: location.address.errors[:address_line_2] %>
       <%= a.text_field :address_line_2, class: 'input-md-3 form-control t-address-line-2' %>
     </div>
 
-    <div class="form-group">
+    <div class="form-group" id="address_line_3">
+      <%= render 'error', error_messages: location.address.errors[:address_line_3] %>
       <%= a.text_field :address_line_3, class: 'input-md-3 form-control t-address-line-3' %>
     </div>
 
-    <div class="form-group">
+    <div class="form-group" id="town">
       <%= a.label :town %>
+      <%= render 'error', error_messages: location.address.errors[:town] %>
       <%= a.text_field :town, class: 'input-md-3 form-control t-town' %>
     </div>
 
-    <div class="form-group">
+    <div class="form-group" id="county">
       <%= a.label :county %>
+      <%= render 'error', error_messages: location.address.errors[:county] %>
       <%= a.text_field :county, class: 'input-md-3 form-control t-county' %>
     </div>
 
-    <div class="form-group">
+    <div class="form-group" id="postcode">
       <%= a.label :postcode %>
+      <%= render 'error', error_messages: location.address.errors[:postcode] %>
       <%= a.text_field :postcode, class: 'input-md-2 form-control t-postcode' %>
     </div>
   <% end %>
 
-  <div class="form-group">
+  <div class="form-group" id="hours">
     <%= f.label :hours, 'Booking hours' %>
+    <%= render 'error', error_messages: location.errors[:hours] %>
     <%= f.text_area :hours, class: 'input-md-3 form-control t-booking-hours' %>
```

Comment:
Are these id's unique? Wouldn't the `text_area` helper generate something with the same ID?